### PR TITLE
CI: add GH Actions workflow for Darwin unit tests

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <mayankshah1614@gmail.com>

**What this PR does / why we need it**:
Add GH Actions workflow for running unit tests on darwin

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: .github/workflows/darwin.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    branches: [ master ]
+jobs:
+  build:
+    name: Unit Tests
+    runs-on: macos-latest
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.13
```
Comment (msau42) [inline .github/workflows/darwin.yaml:15]: Can we use go1.15?

**Review by msau42 — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

File: .github/workflows/darwin.yaml (hunk 1 above)
Comment (andyzhangx) [reply .github/workflows/darwin.yaml:15]: pls follow this up later @mayankshah1607 thanks.

**Pull request merged**